push.sh

```diff
@@ -1,16 +1,66 @@
 #! /bin/sh
 
-# Use WEBDAV_USERNAME or WEBDAV_PASSWORD as default, if provided
+# Use WEBDAV_USERNAME as default, if provided.
 if [ -z "$PLUGIN_USERNAME" ] && [ ! -z "$WEBDAV_USERNAME" ]; then
+
     PLUGIN_USERNAME="$WEBDAV_USERNAME"
 fi
+
+# Use WEBDAV_PASSWORD as default, if provided.
 if [ -z "$PLUGIN_PASSWORD" ] && [ ! -z "$WEBDAV_PASSWORD" ]; then
+
     PLUGIN_PASSWORD="$WEBDAV_PASSWORD"
 fi
 
 # If username and password are provided, add auth
 if [ ! -z "$PLUGIN_USERNAME" ] && [ ! -z "$PLUGIN_PASSWORD" ]; then
-    AUTH="-u ${PLUGIN_USERNAME}:${PLUGIN_PASSWORD}"
+
+    AUTH="--user ${PLUGIN_USERNAME}:${PLUGIN_PASSWORD}"
+fi
+
+# Use a proxy, if one is specified
+if [ ! -z "$PLUGIN_PROXY_URL" ]; then
+
+    PLUGIN_PROXY_URL="--proxy ${PLUGIN_PROXY_URL}"
+fi
+
+# If a timeout is specified, make use of it.
+if [ ! -z "$PLUGIN_TIMEOUT" ]; then
+
+    PLUGIN_TIMEOUT="--max-time ${PLUGIN_TIMEOUT}"
```

Comment:
Now that I'm thinking about it... the value of `${PLUGIN_TIMEOUT}` should be wrapped in quotes in case they pass a space separated string.

Eg.

    PLUGIN_TIMEOUT="--max-time '${PLUGIN_TIMEOUT}'"

Or you can use `\"`.